src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexRunnerFactory.cs

```diff
@@ -93,15 +101,31 @@ protected override void Go()
                 ReadOnlySpan<char> inputSpan = runtext.AsSpan(beginning, runtextend - beginning);
 
                 // Perform the match.
-                SymbolicMatch pos = _matcher.FindMatch(quick, inputSpan, runtextpos - beginning);
+                SymbolicMatch pos = _matcher.FindMatch(quick, inputSpan, runtextpos - beginning, _perThreadData);
 
                 // Transfer the result back to the RegexRunner state.
                 if (pos.Success)
                 {
                     // If we successfully matched, capture the match, and then jump the current position to the end of the match.
                     int start = pos.Index + beginning;
                     int end = start + pos.Length;
-                    Capture(0, start, end);
+                    if (pos.CaptureStarts != null)
```

Comment:
Should this logic also check if quick != true? If we are just doing a quick IsMatch operation, even if the pattern has multiple catpures, I wonder if we should skip this section as well.

Reply:
As an aside, in case we are not doing so already, we should perhaps just skip calculating all of the captures start and ends if quick is set to true